lib/max/Delivery/log.php

```diff
@@ -55,6 +55,18 @@ function MAX_Delivery_log_logAdImpression($adId, $zoneId)
     // Only log impressions if impression logging is enabled
     if (empty($GLOBALS['_MAX']['CONF']['logging']['adImpressions'])) { return true; }
 
+        // If no-impression/click/redirect on inactive banners is enabled,
+        // check to see if the banner is active. If not, exit impression logging
+        // at this point, without recording the impression
+        if (isset($GLOBALS['_MAX']['CONF']['logging']['blockInactiveBanners'])) {
+            $aAdInfo = MAX_cacheGetAd($adId);
```

Comment:
Delivery logging will be slightly slower because blockInactiveBanners is on by default. Would it be worth to keep the old behaviour by default?

Reply:
I actually think the new setting should be enabled by default, even though it is slower. I think that for new, non-power users (the majority of new users?), this is the behaviour that they expect - if a banner is not active, you should not see impressions or clicks recorded.